### DISCUSSION TITLE
field array - validation with yup on change

### DISCUSSION THREAD

**Discussion by mrsekut** [react-hook-form/react-hook-form#7627] [Q&A] [answered]
I want to create a field that meets the following requirements using `useFieldArray`
- ① Validation of each field (enter with 10 characters or less)
- ② Validation of both fields (Enter both or do not enter both)

This time I'm using yup and I'm using `test()` to satisfy ②
However, even though `onChange` is specified, validation does not work at the timing of `onChange`.

prepared a code sandbox.
[code sandbox
](https://codesandbox.io/s/react-hook-form-fieldsarray-yup-validation-min-length-forked-nmc33?file=/src/index.js)

The yup schema itself meets the requirements, but does not partially respond to onChange.

Specifically, the requirements you want to meet are as follows
- [ ] `text1` validation works with onChange
- [ ] `text2` validation works with onChange
- [ ] Validation of both (`test()`) works with onChange
- [ ] Validated at the time of submit (Up to 3 errors)
- [ ] `text1` validation works with onChange (revalidate)
- [ ] `text2` validation works with onChange (revalidate)
- [ ] Validation of both (`test()`) works with onChange(revalidate)

When actually tested with the above Code Sandbox, it will be as follows

#### When commenting out `test()` in the code sandbox example above
- [x] `text1` validation works with onChange
- [x] `text2` validation works with onChange
- [ ] Validation of both (`test()`) works with onChange
- [x] Validated at the time of submit (Up to 3 errors)
- [x] `text1` validation works with onChange (revalidate)
- [x] `text2` validation works with onChange (revalidate)
- [ ] Validation of both (`test()`) works with onChange(revalidate)

#### When using `test()` in the code sandbox example above
- [ ] `text1` validation works with onChange
- [x] `text2` validation works with onChange
- [ ] Validation of both (`test()`) works with onChange
- [x] Validated at the time of submit (Up to 3 errors)
- [ ] `text1` validation works with onChange (revalidate)
- [x] `text2` validation works with onChange (revalidate)
- [ ] Validation of both (`test()`) works with onChange(revalidate)

Is there any way to solve it? (all checked)
- Use trigger?
- Define yup schema without using `test()`?

Please advise.

**Comment by Moshyfawn** [comment, marked as answer]
Take a look at [this CSB](https://codesandbox.io/s/brave-ives-2s91b?file=/src/App.tsx). I don't usually use Yup, so I've modified the validation schema approach to what I can understand. Hope it's still valid for you.
Very nice description of your use case, thank you for the time you took to get your point across

**Reply by mrsekut** [reply]
Completely resolved.
The `noSortEdges` option was completely out of sight.
It was very helpful. I'm really thankful to you.

**Reply by Moshyfawn** [reply]
No problem ;)
